examples/src/streaming/client.rs

```diff
@@ -0,0 +1,133 @@
+//! Audio client for client-driven streaming
+
+use super::protocol::*;
+use std::collections::VecDeque;
+use std::time::Instant;
+
+/// Client-side audio buffer manager
+#[derive(Debug)]
+pub struct AudioBuffer {
+    chunks: VecDeque<Vec<u8>>,
```

Comment:
Can we use fixed size channels for buffering? like we have a loop that keeps writing to channel, fixed size back-pressure.